src/test/scala/is/hail/variant/vsm/VSMSuite.scala

```diff
@@ -52,115 +52,140 @@ class VSMSuite extends SparkSuite {
     val v2 = Variant("1", 2, "T", "G")
     val v3 = Variant("1", 2, "T", "A")
 
-    val r1 = Annotation(Annotation("yes"), "yes")
-    val r2 = Annotation(Annotation("yes"), "no")
-    val r3 = Annotation(Annotation("no"), "yes")
-
-
-    val va1 = r1
-    val va2 = r2
-    val va3 = r3
-
-    val rdd1 = sc.parallelize(Seq((v1, (va1,
+    val s3rdd1 = sc.parallelize(Seq((v1, (s3va1,
       Iterable(Genotype(),
         Genotype(0),
         Genotype(2)))),
-      (v2, (va2,
+      (v2, (s3va2,
         Iterable(Genotype(0),
           Genotype(0),
           Genotype(1)))))).toOrderedRDD
 
     // differ in variant
-    val rdd2 = sc.parallelize(Seq((v1, (va1,
+    val s3rdd2 = sc.parallelize(Seq((v1, (s3va1,
       Iterable(Genotype(),
         Genotype(0),
         Genotype(2)))),
-      (v3, (va2,
+      (v3, (s3va2,
         Iterable(Genotype(0),
           Genotype(0),
           Genotype(1)))))).toOrderedRDD
 
     // differ in genotype
-    val rdd3 = sc.parallelize(Seq((v1, (va1,
+    val s3rdd3 = sc.parallelize(Seq((v1, (s3va1,
       Iterable(Genotype(),
         Genotype(1),
         Genotype(2)))),
-      (v2, (va2,
+      (v2, (s3va2,
         Iterable(Genotype(0),
           Genotype(0),
           Genotype(1)))))).toOrderedRDD
 
     // for mdata2
-    val rdd4 = sc.parallelize(Seq((v1, (va1,
+    val s2rdd4 = sc.parallelize(Seq((v1, (s2va1,
       Iterable(Genotype(),
         Genotype(0)))),
-      (v2, (va2, Iterable(
+      (v2, (s2va2, Iterable(
         Genotype(0),
         Genotype(0)))))).toOrderedRDD
 
     // differ in number of variants
-    val rdd5 = sc.parallelize(Seq((v1, (va1,
+    val s3rdd5 = sc.parallelize(Seq((v1, (s3va1,
       Iterable(Genotype(),
         Genotype(0)))))).toOrderedRDD
 
     // differ in annotations
-    val rdd6 = sc.parallelize(Seq((v1, (va1,
+    val s3rdd6 = sc.parallelize(Seq((v1, (s3va1,
       Iterable(Genotype(),
         Genotype(0),
         Genotype(2)))),
-      (v2, (va3,
+      (v2, (s3va3,
         Iterable(Genotype(0),
           Genotype(0),
           Genotype(1)))))).toOrderedRDD
 
-    val vdss = Array(new VariantDataset(hc, mdata1, rdd1),
-      new VariantDataset(hc, mdata1, rdd2),
-      new VariantDataset(hc, mdata1, rdd3),
-      new VariantDataset(hc, mdata2, rdd1),
-      new VariantDataset(hc, mdata2, rdd2),
-      new VariantDataset(hc, mdata2, rdd3),
-      new VariantDataset(hc, mdata2, rdd4),
-      new VariantDataset(hc, mdata2, rdd5),
-      new VariantDataset(hc, mdata3, rdd1),
-      new VariantDataset(hc, mdata3, rdd2),
-      new VariantDataset(hc, mdata4, rdd1),
-      new VariantDataset(hc, mdata4, rdd2),
-      new VariantDataset(hc, mdata1, rdd6))
+    val s1rdd7 = sc.parallelize(Seq((v1, (s1va1,
+      Iterable(Genotype(),
+        Genotype(0),
+        Genotype(2)))),
+      (v2, (s1va2,
+        Iterable(Genotype(0),
+          Genotype(0),
+          Genotype(1)))))).toOrderedRDD
+
+    val s2rdd8 = sc.parallelize(Seq((v1, (s2va1,
```

Comment:
Never used.  Did you intend to add `new VariantDataset(hc, s2mdata, s2rdd8)` to the list below?